cli/cmd/init.go

```diff
@@ -221,6 +254,49 @@ func validateConfigurationIntegirty(ctx context.Context, happyClient *HappyClien
 	}
 }
 
+// pull an image from a stack and push it to a new stack
+// an optional tag can be provided to cherry pick a stack's image from the image history
+func pullAndPushImageFrom(
+	ctx context.Context,
+	srcStackName, srcTag string, srcHappyClient *HappyClient,
+	targetHappyClient *HappyClient,
+) error {
+	err := validate(validateStackExistsUpdate(ctx, srcStackName, srcHappyClient))
+	if err != nil {
+		return errors.Wrapf(err, "stack %s doesn't exist", srcStackName)
+	}
+	// if no tag is specified, get the latest deployed tag
+	// by reading the image variable from the TFE workspace
+	if srcTag == "" {
+		var err error
+		srcTag, err = srcHappyClient.StackService.GetLatestDeployedTag(ctx, srcStackName)
+		if err != nil {
+			return errors.Wrapf(err, "unable to get latest tag from stack %s", srcStackName)
+		}
+	} else {
+		exists, err := srcHappyClient.ArtifactBuilder.CheckImageExists(ctx, srcTag)
+		if err != nil {
+			return errors.Wrapf(err, "error checking if tag %s existed", srcTag)
+		}
+		if !exists {
+			return errors.Errorf("src image tag does not exist %s:%s", srcStackName, srcTag)
+		}
+	}
+
+	servicesImage, err := srcHappyClient.ArtifactBuilder.Pull(ctx, srcStackName, srcTag)
+	if err != nil {
+		return errors.Wrapf(err, "unable to pull image %s from stack %s in env %s", srcTag, srcStackName, srcHappyClient.HappyConfig.GetEnv())
+	}
+	err = targetHappyClient.ArtifactBuilder.PushFromWithTag(ctx, servicesImage, srcTag)
```

Comment:
does this actually pull/push images to/from different ecr repos? Is that necessary?

Reply:
yup it does pull and push them. I suppose we could get away with not doing that and just point to the original image, but I think it doesn't really cost anything to have an extra copy and isolates the image in case the original one is ever torn down 🤷🏻

Reply:
There are tradeoffs either way, though I think I'd slightly prefer to support external repos. This works for now, but our CI roles are going to need access to both repos now to allow a "pull" from a dev/staging ecr and "push" to a prod ecr.